### PR TITLE
Fix leaktests

### PR DESCRIPTION
**Pull request by Therzok**

### CODE REVIEW

File: main/src/addins/MonoDevelop.DesignerSupport/MonoDevelop.DesignerSupport/DesignerSupportService.cs [hunk 1]
```diff
@@ -249,6 +249,13 @@ internal DesignerSupportService ()
 		{
 			IdeApp.CommandService.RegisterCommandTargetVisitor (new PropertyPadVisitor ());
 			AddinManager.ExtensionChanged += OnExtensionChanged;
+
+			IdeApp.Initialized += (s, args) => {				
+				IdeApp.Workbench.ActiveDocumentChanged += (o, args) => {
+					if (lastPadProvider != null && args.Document == null)
```
Comment (netonjm) [inline main/src/addins/MonoDevelop.DesignerSupport/MonoDevelop.DesignerSupport/DesignerSupportService.cs]: @Therzok  we should check also lastCustomProvider ?https://github.com/mono/monodevelop/blob/98f87c05c062d06d49ca683265fca198435aa8ea/main/src/addins/MonoDevelop.DesignerSupport/MonoDevelop.DesignerSupport/DesignerSupportService.cs#L52

Comment (Therzok) [reply main/src/addins/MonoDevelop.DesignerSupport/MonoDevelop.DesignerSupport/DesignerSupportService.cs]: We could.

**Comment by Therzok** [comment]
@monojenkins backport release-8.4

**Comment by Therzok** [comment]
Debugger failure unrelated: https://github.com/mono/monodevelop/pull/8956

**Comment by Therzok** [comment]
@monojenkins backport release-8.4-preview1

**Pull request merged**